packages/commonwealth/client/scripts/stores/ProposalStore.ts

```diff
@@ -5,6 +5,11 @@ class ProposalStore<ProposalT extends IIdentifiable> extends Store<ProposalT> {
   private _storeId: { [hash: string]: ProposalT } = {};
 
   public add(proposal: ProposalT) {
+    const existingProposal = this.getByIdentifier(proposal.identifier);
+    if (existingProposal) {
+      this.remove(existingProposal);
+    }
+
```

Comment:
@jnaviask Can you think of any possible side-effects here, maybe in other communities?

Reply:
Nope, this should be fine.